li-protobuf/src/main/java/com/linkedin/data/protobuf/ProtoWriter.java

```diff
@@ -137,7 +139,19 @@ public void writeBytes(byte[] value, int offset, int length) throws IOException
   }
 
   /**
-   * Write a 32-bit signed integer.
+   * Write a fixed length 32-bit signed integer.
+   */
+  public final void writeFixedInt32(final int value) throws IOException
+  {
+    flushIfNotAvailable(FIXED32_SIZE);
```

Comment:
do we need this? writing size can be skipped here right? same with int64

Reply:
We are not writing size here. We are flushing the buffer if needed and making sure that at least 4 bytes in the case of int and 8 bytes in the case of double are available to write in the buffer.